wallet/createtx.go

```diff
@@ -101,15 +101,25 @@ func (s secretSource) GetScript(addr btcutil.Address) ([]byte, error) {
 // UTXO set and minconf policy. An additional output may be added to return
 // change to the wallet.  An appropriate fee is included based on the wallet's
 // current relay fee.  The wallet must be unlocked to create the transaction.
+//
+// NOTE: The dryRun argument can be set true to create a tx that doesn't alter
+// the database. A tx created with this set to true SHOULD NOT be broadcasted.
 func (w *Wallet) txToOutputs(outputs []*wire.TxOut, account uint32,
-	minconf int32, feeSatPerKb btcutil.Amount) (tx *txauthor.AuthoredTx, err error) {
+	minconf int32, feeSatPerKb btcutil.Amount, dryRun bool) (
+	tx *txauthor.AuthoredTx, err error) {
 
 	chainClient, err := w.requireChainClient()
 	if err != nil {
 		return nil, err
 	}
 
 	err = walletdb.Update(w.db, func(dbtx walletdb.ReadWriteTx) error {
+		// If dry run is requested, we must make sure the database is
+		// not altered.
+		if dryRun {
```

Comment:
Ideally we add a test to ensure that this works properly. At glance it _seems_ that it should, but would like an additional level of certainty.

Reply:
Very good idea, turned out writing a test surfaced all kinds of problems... To be continued